File: dress-design-backend/src/main/java/com/fashion/dto/Customer.java
package com.fashion.dto;

import java.time.LocalDateTime;
import java.util.UUID;

import javax.persistence.Id;

import org.springframework.data.mongodb.core.mapping.Document;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateTimeDeserializer;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Document(collection = "customers")
@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Builder
public class Customer {

	@Id
	private UUID id;

	private String customerId;

	private String customerName;

	private String phone;

	private String email;

	private String address;

	private int version;
	
	@JsonDeserialize(using = LocalDateTimeDeserializer.class)
	private LocalDateTime createdAt;

	private String createdByUserId;
	
	@JsonDeserialize(using = LocalDateTimeDeserializer.class)
	private LocalDateTime updatedAt;
	
}

File: dress-ui/src/app/models/customer.ts
export interface Customer {
    id: string,
    customerId: string,
    customerName: string,
    phone: string,
    email: string,
    address: string,
    status: boolean,
    version: number,
    createdAt: string,
    createdByUserId: string,
    updatedAt: string
}

export interface AddCustomer {
    customerName: string,
    phone: string,
    email: string,
    address: string,
}